lib/repository.py

```diff
@@ -35,12 +36,15 @@ def __init__(self, push_info):
         super(PushError, self).__init__(message)
 
 
-def get_git_repository(name, remote_repo_url, parent_dir_path):
+def get_git_repository(remote_repo_url, parent_dir_path):
     """
-    Get a local git repository located in a subdirectory of the parent
-    dir, according to specified name.
+    Get a local git repository located in a subdirectory of the parent directory.
     If it does not exist, clone it from the specified URL.
     """
+    # infer name
+    url_parts = urlparse.urlparse(remote_repo_url)
+    name = os.path.basename(url_parts.path)
```

Comment:
I think that `# infer name` could be removed.

It doesn't add much information to that context. It ends up being just noise in the source code.